src/Analyzers/CSharp/Analyzers/OrderModifiers/CSharpOrderModifiersDiagnosticAnalyzer.cs

```diff
@@ -30,29 +30,22 @@ public CSharpOrderModifiersDiagnosticAnalyzer()
             ReportDiagnostic severity,
             SyntaxNode root)
         {
-            foreach (var child in root.ChildNodesAndTokens())
+            foreach (var node in root.ChildNodes())
             {
-                if (child.IsNode)
+                if (node is MemberDeclarationSyntax || node.IsKind(SyntaxKind.LocalFunctionStatement))
                 {
-                    var node = child.AsNode();
-                    if (node is MemberDeclarationSyntax memberDeclaration)
-                    {
-                        CheckModifiers(context, preferredOrder, severity, memberDeclaration);
-
-                        // Recurse and check children.  Note: we only do this if we're on an actual 
-                        // member declaration.  Once we hit something that isn't, we don't need to 
-                        // keep recursing.  This prevents us from actually entering things like method 
-                        // bodies.
-                        Recurse(context, preferredOrder, severity, node);
-                    }
-                    else if (node is AccessorListSyntax accessorList)
+                    CheckModifiers(context, preferredOrder, severity, node);
+                }
+                else if (node is AccessorListSyntax accessorList)
+                {
+                    foreach (var accessor in accessorList.Accessors)
                     {
-                        foreach (var accessor in accessorList.Accessors)
-                        {
-                            CheckModifiers(context, preferredOrder, severity, accessor);
-                        }
+                        CheckModifiers(context, preferredOrder, severity, accessor);
                     }
                 }
+
+                // We don't stop at member declarations only because this prevents us from visiting local functions.
+                Recurse(context, preferredOrder, severity, node);
```

Comment:
this is a *lot* of recursing.  the original code was intentional in letting us bail out almost immediately.  we should attemp to preserve that.  

Note: i don't know how you could order modifiers for locals currently.  so consider undoing this part.

Reply:
@CyrusNajmabadi Should I attempt to find an in-between situation? i.e, do less recursion than what I currently have, but would be more than what previously existed to be able to handle local functions.

Reply:
i'm not sure what in-between option woudl exist.  local functions can be inside lambdas.  So we'd need to dive arbitrarily deep into statements/expressions to find these :-/

Reply:
@CyrusNajmabadi I was thinking of stop recursing if the node isn't AccessorList, MemberDeclaration, or *BlockSyntax* (the extra one I'll add).

Reply:
Right. But that wouldn't help if you have a local function inside a lambda. 

Does this *need* to have it's behavior changed?  I get there is a deficiency here. But if it's not bothering people, I would prefer we just not change this feature.

Reply:
> Right. But that wouldn't help if you have a local function inside a lambda.

Not sure why? The local function would still be a child of a BlockSyntax.

![image](https://user-images.githubusercontent.com/31348972/113267346-8c557700-92d6-11eb-92c0-ba19557208b0.png)

Reply:
To find the block, you'd have to recurse through non blocks.

Reply:
@CyrusNajmabadi Does using `DescendantNodes` for the specific nodes we're looking for make things any better? If not, I'm going to revert.

Reply:
Nope.  It will examine *everything*.  :(